MIGRATION_GUIDE.md

```diff
@@ -1,4 +1,325 @@
-# Migration Guide: v4 to v5
```

Comment:
Maybe we could use `diff` blocks to make migrations easier?

```diff
- old
+ new
```

Reply:
I think that makes it a little harder to read in some cases, I'm just gonna leave it for now